src/Domain/Handler/SubscriptionRefundedEventHandler.ts

```diff
@@ -46,15 +48,22 @@ implements DomainEventHandlerInterface
   }
 
   private async updateUserRole(user: User): Promise<void> {
-    const role = await this.roleRepository.findOneByName(RoleName.CoreUser)
+    const currentRoleName = (await user.roles)[0].name as RoleName
+    const newRoleName = RoleName.CoreUser
```

Comment:
hmm this should be fetched from the event right? Which for now is indeed populated with the core user role

Reply:
Yeah makes sense! Not sure why I thought the subcription name in the event would correspond to the plan being refunded

Reply:
@karolsojko since after this change updating the user role would behave in the same way for both the purchase and the refund event and we would also need the subscription name to role name map in both, to keep it DRY I've extracted these to a separate common service, the RoleService. Let me know what you think 🙂

Reply:
Perfetto 👌